apps/wallet/tests/staking.spec.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { expect, test } from './fixtures';
+import { createWallet } from './utils/auth';
+
+const TEST_TIMEOUT = 65 * 1000;
+const STAKE_AMOUNT = 100;
+
+test('staking', async ({ page, extensionUrl }) => {
+	test.setTimeout(TEST_TIMEOUT);
+
+	await createWallet(page, extensionUrl);
+
+	await page.getByTestId('faucet-request-button').click();
+	await expect(page.getByTestId('coin-balance')).not.toHaveText('0SUI');
+
+	await page.getByTestId('stake-and-earn-button').click();
+	await page.getByTestId('validator-list-item').first().click();
+	await page.getByTestId('select-validator-cta').click();
+	await page.getByTestId('stake-amount-input').fill(STAKE_AMOUNT.toString());
+	await page.getByRole('button', { name: 'Stake Now' }).click();
+	await expect(page.getByTestId('loading-indicator')).not.toBeVisible({
+		timeout: TEST_TIMEOUT,
+	});
+	await expect(page.getByText(/Stake failed/i)).not.toBeVisible();
```

Comment:
Can you make this a positive assertion instead of a negative assertion?

Reply:
Not sure what else to look for here. The page goes to `Transaction` once it is done (which is the next step). This is used to gate the next step if it's failed and we can know that the staking has failed.